internal/service/resources/api/server.go

```diff
@@ -129,13 +133,40 @@ func (r *ResourceServer) GetSubscriptions(ctx context.Context, request api.GetSu
 	return api.GetSubscriptions200JSONResponse(objects), nil
 }
 
+// validateSubscription validates a subscription before accepting the request
+func (r *ResourceServer) validateSubscription(request api.CreateSubscriptionRequestObject) error {
```

Comment:
I kept the name generic to later add in validation of the "filter" once that's defined.   A bit difficult to make this a common function since the alarm subscription and inventory subscription are technically different objects, and as it stands (and unfortunately) their "filter" definitions are not the same.... we're waiting for conformation about whether they are truly different or if there is a mistake somewhere.